syntax="proto3";
//注册
message ReqSignup
{
    string username = 1;
    string password = 2;
}

message RespSignup
{
    int32 code = 1;
    string msg = 2;
}

service SignupService 
{
    rpc Signup(ReqSignup) returns (RespSignup) {}
}

//登录
message ReqSignin
{
    string username = 1;
    string password = 2;
}

message RespSignin
{
    int32 code = 1;
    string msg = 2;
    optional string token = 3;
    optional string username = 4;
    optional string location = 5;
}
service SigninService 
{
    rpc Signin(ReqSignin) returns (RespSignin) {}
}

//用户信息
message ReqUserinfo
{
    string username = 1;
    string token = 2;
}

message RespUserinfo
{
    int32 code = 1;
    string msg = 2;
    optional string username = 3;
    optional string signupat = 4;
}

service UserinfoService
{
    rpc Userinfo(ReqUserinfo) returns (RespUserinfo) {}
}
